.github/workflows/canary.yml

```diff
@@ -27,17 +27,18 @@ jobs:
         with:
           ref: ${{ env.HEAD_COMMIT }}
       - name: Set up Node
-        uses: actions/setup-node@v3
-        with:
-          # ember won't build under node 16, at least not with the versions of the ember build tools we use
-          node-version: 14
+        uses: volta-cli/action@v4
       - name: Install dependencies
         run: yarn install --ignore-engines --frozen-lockfile
       - name: Build packages
         run: yarn build
+      - name: Get node version
+        id: versions
+        run: |
+          echo "echo node=$(jq -r '.volta.node' package.json)" >> $GITHUB_OUTPUT
       - name: Run Canary Tests
         env:
-          E2E_TEST_PUBLISH_SCRIPT_NODE_VERSION: ${{ env.DEFAULT_NODE_VERSION }}
+          E2E_TEST_PUBLISH_SCRIPT_NODE_VERSION: ${{ steps.versions.outputs.node }}
```

Comment:
@lforst Do we actually need to set this, when it is the default installed node version? 🤔 Just wondering, because in the canary tests it was actually also setting it to `env.DEFAULT_NODE_VERSION`, which isn't actually set there 😅  (same applies then to the non-canary tests I guess)

Reply:
I can't speak to the end-to-end tests, but as for the others, GH recommends always being explicit, so I think it's not a bad thing to do.

Reply:
We use this env variable to set what node version we use to fake publish our packages in the E2E tests. The version should always align with the version we actually publish our packages with. If in the future we want to do our E2E tests with multiple node versions this will be necessary. Here is how this came to be: https://github.com/getsentry/sentry-javascript/pull/5806#discussion_r979767950 Looking back at it now, I probably wouldn't do it again this way. Maybe we can simplify.

As for the canary tests: Looks like I forgot to copy the env var over. (Which wasn't too bad since right now we only tested with the default node version anyways but good that you caught it.)

Reply:
makes sense. So we can leave this in (this parses the node version defined for volta, so it is still consistent, but we can change it later if needed)!